code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/URLUtils.java

```diff
@@ -0,0 +1,658 @@
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		false,
+		false,
+		false,
+		false,
+		true,
+		false,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		true,
+		false,
+		false,
+		false,
+		true,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+		false,
+	};
+
+	/**
+	 * Helper function to generate url variables in format acceptable by the AEP web SDKs
+	 *
+	 * @param ts timestamp {@link String} denoting time when url variables request was made
+	 * @param ecid Experience Cloud identifier {@link String} generated by the SDK
+	 * @param orgId Experience Cloud Org identifier {@link String} set in the configuration
+	 * @return {@link String} formatted with the visitor id payload
+	 */
+	static String generateURLVariablesPayload(final String ts, final String ecid, final String orgId) {
+		final StringBuilder urlFragment = new StringBuilder();
+
+		// construct the adobe_mc string
+
+		// append timestamp
+		String theIdString = appendKVPToVisitorIdString(null, IdentityConstants.UrlKeys.TS, ts);
+
+		// append ecid
+		theIdString = appendKVPToVisitorIdString(theIdString, IdentityConstants.UrlKeys.EXPERIENCE_CLOUD_ID, ecid);
+
+		// add Experience Cloud Org ID
+		theIdString = appendKVPToVisitorIdString(theIdString, IdentityConstants.UrlKeys.EXPERIENCE_CLOUD_ORG_ID, orgId);
+
+		// after the adobe_mc string is created, encode the idString before adding it to the url
+		urlFragment.append(IdentityConstants.UrlKeys.PAYLOAD);
+		urlFragment.append("=");
+
+		String encodedIdString = urlEncode(theIdString);
+		if (encodedIdString != null) {
```

Comment:
Remove this null check. If the result is null and you don't append, then the returned string is "adobe_mc=" which can mess up a customer's URL. If a null result is appended, like it is done in Identity direct, then the result is "adobe_mc=null", which provides a value to the query parameter.

Reply:
This will not be returned to the customer, we handle missing orgId, ecid in our extension code. I removed all the checks from utils to keep it lighter.